circuit_maintenance_parser/providers.py

```diff
@@ -73,12 +76,15 @@ def process(self, data_type: str = "") -> Iterable[Maintenance]:
                 )
                 if data_type and data_type == parser.get_data_type() or not data_type:
                     return parser.process()
-            except (ParsingError, MissingMandatoryFields):
+            except (ParsingError, MissingMandatoryFields) as exc:
                 logger.debug(
-                    "Parser %s for provider %s was not successful", parser_class.__name__, self.__class__.__name__
+                    "Parser %s for provider %s was not successful: %s", parser_name, provider_name, exc,
```

Comment:
Given that this is a `debug` log, I'd personally prefer it to contain the full traceback, something like:

```python
logger.debug(
    "Parser %s for provider %s was not successful:\n%s",
    parser_name,
    provider_name,
    traceback.format_exc(),
)
```